### PR TITLE
Fast event idx lookup, dynamic spectrum range

### PR DESCRIPTION
**Pull request by marcomuzio**
Added fast event index lookup for a requested event number. This is more efficient than a for loop to find the ROOT file index corresponding to a given event number. Also made spectrum plots y range dynamic.

Finally, added some functionality for if we ever want to check the station id that's stored in the ROOT data (stored as a class variable `data_station_id`. In the ended I decided not to use it, but I already coded this up and maybe it will be useful later. Can remove if wanted.

### CODE REVIEW

File: araproc/framework/data_visualization.py [hunk 1]
```diff
@@ -81,6 +82,8 @@ def plot_waveform_bundle(
             xvals = freqs
             yvals = np.log10(np.abs(spectrum))
```
Comment (Mohammad-Neutrino) [inline araproc/framework/data_visualization.py:83]: Thanks Marco for making things more dynamic. Now that you have opened this PR, could you kindly make the power axis in terms of **dBm** and frequency in **MHz**? I think 
```suggestion
            xvals = freqs*1e3 # from GHz to MHz
            yvals = 10*np.log10(np.abs(spectrum)**2 / 50 / 1e3) # from mV to dBm
```
will do the job.

Comment (Mohammad-Neutrino) [reply araproc/framework/data_visualization.py:83]: Accordingly, we can change 
line 44 to
```suggestion
           "freq" : "Frequency (MHz)"
```
line 48 to 
```suggestion
           "freq" : "Spectrum (dBm)"
```

File: araproc/framework/data_visualization.py [hunk 2]
```diff
@@ -92,7 +95,7 @@ def plot_waveform_bundle(
     
     # make limits look nice
     if time_or_freq == "freq":
-        ax.set_ylim([1,4])
+        ax.set_ylim([1,ymax+0.2])
```
Comment (Mohammad-Neutrino) [inline araproc/framework/data_visualization.py:98]: Also ylim of `ax.set_ylim(-50, 50)` should catch the whole y axis in `dBm` but you can go with `-50 to ymax+1` or something similar for `ymin` as `ymin-1`

File: araproc/framework/dataset.py [hunk 3]
```diff
@@ -144,6 +144,7 @@ def __init__(self,
         self.event_tree = None
         self.run_number = None
         self.station_id = None
+        self.data_station_id = None
```
Comment (Mohammad-Neutrino) [inline araproc/framework/dataset.py:147]: I like the idea of keeping the `stationID`. Thanks for adding that!

**Comment by marcomuzio** [comment]
Just fyi @Mohammad-Neutrino decided to switch the spectrum frequency axis back to GHz since it's the natural unit for AraProc and also our filters and stuff are in GHz. So hopefully will minimize confusion.

**Pull request merged**